### PR TITLE
Initial desktop support

### PR DESCRIPTION
**Pull request by bdlukaa**
Initial desktop support

- Created `packages/unity_video_player`,  which contains the platform-specific code for video playback
  - For Linux and macOS, [dart_vlc](https://pub.dev/packages/dart_vlc) is used
  - For Windows, [media_kit](https://github.com/alexmercerind/media_kit) is used
  - For Android and iOS, [fijkplayer](https://pub.dev/packages/fijkplayer) is used
- Added adaptive navigation:
  - On small screens (< 640 px - phones), minimal side navigation is used (drawer).
  - On wide screens (640 px > - tablets), compact side navigation is used (compact navigation rail).
  - On extra wide screens (1008 px - desktop), top navigation is used
- Added directionality support (right to left / left to right - https://api.flutter.dev/flutter/widgets/Directionality-class.html)
- Added layout funcitonality:
  - Possibility to create multiple layouts
  - Cycle for layouts every 5 seconds, when enabled
  - Multiple layout types: ¹single view, ²multiple view and ³compact view
- Created `/installer`, which contains info about the app installer (windows-only)

For desktop:
image

For tablets:
image

### CODE REVIEW

**Review by alexmercerind — APPROVED** [review]
I'm not asking to merge this for now, because any regressions can result in bugs for existing iOS & Android implementations (which are automatically built using GH actions). 
However, from early look this looks fine & any progress in the direction is good. Surely, more work will be needed to be as feature-proof as existing desktop client.

I'm occupied with exams till December 15th & will be able to tell better afterwards. I will also proceed with replacing package:dart_vlc with new solution. I also have a new window frame library, which I will replace package:window_manager with. It'll also support Windows 7 etc. etc.

**Comment by alexmercerind** [comment]
Maybe addressing @curtishall's feedback directly will be a good idea. He can personally try out the current work done for Windows, in my absence.

Thanks!

**Comment by bdlukaa** [comment]
> any regressions can result in bugs for existing iOS & Android implementations

I agree!

As soon as I added `dart_vlc` as a direct dependency, I realized it'd affect the mobile side (it'd add unnecessary dependencies). To counterpart this issue, I believe it'd be better to create a local federated package to handle video playback - and other platform-specific features.

I'll mark this back as a draft because ¹ of the above issue and ² I'd like to implement all features before merging

File: packages/unity_video_player/unity_video_player_media_kit/lib/unity_video_player_media_kit.dart [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  Future<void> release() async {}
+  @override
+  Future<void> reset() async {
+    await pause();
+    await seekTo(Duration.zero);
+  }
+
+  @override
+  void dispose() {
+    mkPlayer.dispose();
```
Comment (alexmercerind) [inline packages/unity_video_player/unity_video_player_media_kit/lib/unity_video_player_media_kit.dart:171]:
```suggestion
    Future.microtask(() async {
      await mkPlayer.dispose();
      await controller?.dispose();
    });
```

`VideoController` seems to be leaking.

File: installer/installer-test.iss [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+Name: "slovenian"; MessagesFile: "compiler:Languages\Slovenian.isl"
+Name: "spanish"; MessagesFile: "compiler:Languages\Spanish.isl"
+Name: "turkish"; MessagesFile: "compiler:Languages\Turkish.isl"
+Name: "ukrainian"; MessagesFile: "compiler:Languages\Ukrainian.isl"
+
+[Tasks]
+Name: "desktopicon"; Description: "{cm:CreateDesktopIcon}"; GroupDescription: "{cm:AdditionalIcons}"; Flags: unchecked
+
+[Files]
+Source: "C:\Users\bruno\Documents\flutter\paid_projects\unity\build\windows\runner\Release\{#MyAppExeName}"; DestDir: "{app}"; Flags: ignoreversion
```
Comment (alexmercerind) [inline installer/installer-test.iss:62]: Maybe relative paths? Or this a mis-commit?

File: packages/unity_multi_window/lib/unity_multi_window.dart [hunk 3]
```diff
@@ -0,0 +1,39 @@
+library unity_multi_window;
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:flutter/foundation.dart';
+
+class MultiWindow {
+  static Future<ResultWindow> run([List<String> arguments = const []]) async {
+    final result = await Process.start(
```
Comment (alexmercerind) [inline packages/unity_multi_window/lib/unity_multi_window.dart:10]: Impressive!

**Review by alexmercerind — COMMENTED** [review]
That's definitely some good work.

**Pull request merged**